it/src/test/java/apoc/it/core/RBACOnLoadTest.java

```diff
@@ -137,6 +142,71 @@ public void testRBACOnDeny() throws IOException {
         }
     }
 
+    @Test
+    public void testRBACOnFileDeny() throws IOException {
+        String url = "http://localhost/file.txt";
+        addRBACOnLoad(url, "test");
+
+        List<String> loadableAPOCProcs = List.of(
+                "apoc.load.json($url)",
+                "apoc.load.jsonArray($url)",
+                "apoc.load.jsonParams($url, null, null, null, {})",
+                "apoc.load.xml($url)",
+                "apoc.load.arrow($url)",
+                "apoc.import.csv([{fileName: $url, labels: ['Person']}], [], {})",
+                "apoc.import.graphml($url, {})",
+                "apoc.import.json($url)",
+                "apoc.import.xml($url)");
+
+        for (String loadableAPOCProc : loadableAPOCProcs) {
+            RuntimeException e = assertThrows(
+                    RuntimeException.class,
+                    () -> testCall(testUserSession, "CALL " + loadableAPOCProc, Map.of("url", url), r -> {}));
+
+            Assertions.assertThat(e.getMessage()).contains("URLAccessValidationError");
+            Assertions.assertThat(e.getMessage())
+                    .contains("Cause: LOAD on URL '" + url
+                            + "' is denied for user 'testUser' with roles [PUBLIC, editor, test] restricted to");
+        }
+    }
+
+    @Test
+    public void testRBACOnFileDenyAll() throws IOException {
+        String url = "file:///dir/file.txt";
+
+        List<String> loadableAPOCProcs = List.of(
+                "apoc.load.json($url)",
+                "apoc.load.jsonArray($url)",
+                "apoc.load.jsonParams($url, null, null, null, {})",
+                "apoc.load.xml($url)",
+                "apoc.load.arrow($url)",
+                "apoc.import.csv([{fileName: $url, labels: ['Person']}], [], {})",
+                "apoc.import.graphml($url, {})",
+                "apoc.import.json($url)",
+                "apoc.import.xml($url)");
+
+        for (String loadableAPOCProc : loadableAPOCProcs) {
+            RuntimeException e = assertThrows(
+                    RuntimeException.class,
+                    () -> testCall(testUserSession, "CALL " + loadableAPOCProc, Map.of("url", url), r -> {}));
+
+            Assertions.assertThat(e.getMessage()).contains("IOException");
```

Comment:
Probably leave out this "working" part, there are other tests for checking these procedures work with files